### PR TITLE
DCO-10410 build: Change how dnf repos are managed.

### PR DESCRIPTION
**Pull request by ashleypittman**
PR-repos: libfabric@PR-61
Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

**Review by raydennis — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]
Could you please add a more detailed explanation of what and why the changes are needed in the description and make sure that that description is used when it's landed, so that anyone looking back at this commit can understand why it was done.

And really, it would be ideal if the corresponding change was made to any other Dockerfiles (el.8 at least -- but a side-by-side comparison would be best -- `vim -d` is great for this) to try to keep them as synchronised as possible.  If you do that, you can just update your commit message in that commit.

**Comment by ashleypittman** [comment]
> Could you please add a more detailed explanation of what and why the changes are needed in the description and make sure that that description is used when it's landed, so that anyone looking back at this commit can understand why it was done.
> 
> And really, it would be ideal if the corresponding change was made to any other Dockerfiles (el.8 at least -- but a side-by-side comparison would be best -- `vim -d` is great for this) to try to keep them as synchronised as possible. If you do that, you can just update your commit message in that commit.

I was a bit hurried last night and didn't initially have high confidence this would work initially.  If the Redhat ticket page is correct the same bug doesn't exist on el.8 however the fix still applies and would be good to have - like you say we put a lot of effort into keeping this files in sync.

Will refresh.

**Review by raydennis — DISMISSED** [review]

File: utils/docker/Dockerfile.centos.7 [hunk 1]
```diff
@@ -33,6 +33,7 @@ RUN set -e;                                                      \
     if [ -n "$REPO_FILE_URL" ]; then                             \
         dnf --quiet config-manager --disable epel;               \
     fi;                                                          \
+    dnf update epel-release;                                     \
```
[the remark this answers is not in the document]
Comment (raydennis) [reply utils/docker/Dockerfile.centos.7:36]: Ah!  Explains why that one failed while this one succeeded.  Thank you!

**Review by jolivier23 — DISMISSED** [review]

**Review by brianjmurrell — DISMISSED** [review]

**Review by raydennis — APPROVED** [review]

**Comment by ashleypittman** [comment]
Please review.  This is the complete fix for the docker deadlock two weeks ago.  We only landed a partial cherry-pick of this PR in a different commit.

**Review by JohnMalmberg — APPROVED** [review]
More fixes to the Dockerfiles are needed for other issues  and can be covered in https://github.com/daos-stack/daos/pull/9174

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**